clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -401,9 +554,164 @@ export class CopybookDownloadService {
+        if (DATASET in config && this.dsnDownloader) {
+          const dsnSuccess = await this.downloadFromPaths(
+            this.dsnDownloader,
+            copybookName,
+            documentUri,
+            [
+              {
+                path: config.dataset,
+                profile: config.profile ? config.profile : undefined,
+              },
+            ],
+          );
+          if (dsnSuccess) return true;
+        } else if (USS in config && this.ussDownloader) {
+          const ussSuccess = await this.downloadFromPaths(
+            this.ussDownloader,
+            copybookName,
+            documentUri,
+            [
+              {
+                path: config.uss,
+                profile: config.profile ? config.profile : undefined,
+              },
+            ],
+          );
+          if (ussSuccess) return true;
+        } else if (ENVIRONMENT in config && this.e4eDownloader) {
+          const resolvedProfile = await this.e4eDownloader.getProfileInfo(
+            config.profile,
+          );
+          const element: EndevorElement = {
+            use_map: config.use_map === false ? false : true,
+            environment: config.environment,
+            stage: config.stage,
+            system: config.system,
+            subsystem: config.subsystem,
+            type: config.type,
+            element: copybookName.name.toUpperCase(),
+            fingerprint: "",
+          };
+          if (
+            resolvedProfile &&
+            (await this.e4eDownloader.hasElement(
+              resolvedProfile,
+              element,
+              copybookName.name,
+            )) &&
+            (await this.e4eDownloader?.downloadElementE4E(
+              resolvedProfile,
+              element,
+            ))
+          )
+            return true;
+        }
+      }
+    } catch (error) {
+      this.outputChannel?.appendLine(getErrorMessage(error));
+      return false;
+    }
+    return false;
+  }
 
   public reenableFailedRequests() {
     this.dsnDownloader?.reenableFailedRequests();
     this.ussDownloader?.reenableFailedRequests();
   }
 }
+async function searchCopybookinProcessorGroups(
+  documentUri: string,
+  copybookName: string,
+  storagePath: string,
+  pgConfigs: (
+    | string
+    | ZoweDatasetConfigModel
+    | ZoweUssConfigModel
+    | EndevorConfigModel
+  )[],
+  e4eDownloader?: CopybookDownloaderForE4E,
+  dsnDownloader?: CopybookDownloaderForDsn,
+  ussDownloader?: CopybookDownloaderForUss,
+): Promise<vscode.Uri | undefined> {
+  let result: vscode.Uri | undefined;
+
+  for (const config of pgConfigs) {
+    let shouldFound = false;
+    let folders: string = "";
+    let extensions: string[] | undefined;
+    if (typeof config === "string") {
+      folders = config;
+      extensions = await SettingsService.getCopybookExtension(documentUri);
+    } else if (ENVIRONMENT in config && e4eDownloader) {
+      const endevorType = DownloadUtil.endevorConfigToType(config);
```

Comment:
It's still an e4e config...
```suggestion
    } else if (ENVIRONMENT in config) {
      if (!e4eDownloader) continue;
      const endevorType = DownloadUtil.endevorConfigToType(config);
```